src/metabase/driver.clj

```diff
@@ -304,7 +304,7 @@
         java.math.BigInteger            :BigIntegerField
         java.sql.Date                   :DateField
         java.sql.Timestamp              :DateTimeField
-        java.util.Date                  :DateField
+        java.util.Date                  :DateTimeField
```

Comment:
Let me poke around on it a bit.  I hope that the frontend is accounting for this because doing an hourly grouping on a `:DateField` should be broken.

You are right that the set of options for base_types is a bit redundant in places, so we could clean it up a bit.